src/Strategy/Dart/PubSpec.hs

```diff
@@ -0,0 +1,178 @@
+module Strategy.Dart.PubSpec (
+  analyzePubSpecFile,
+
+  -- * for testing
+  buildGraph,
+  PubSpecContent (..),
+  PubSpecDepSource (..),
+) where
+
+import Control.Effect.Diagnostics (Diagnostics, context)
+import Data.Foldable (asum, for_)
+import Data.Map (Map, filterWithKey, member, notMember, toList)
+import Data.Map.Strict qualified as Map
+import Data.Maybe (fromMaybe, maybeToList)
+import Data.Text (Text)
+import Data.Yaml (FromJSON (parseJSON), (.:), (.:?))
+import Data.Yaml qualified as Yaml
+import DepTypes (
+  DepEnvironment (EnvDevelopment, EnvProduction),
+  DepType (GitType, PubType),
+  Dependency (..),
+  VerConstraint (CEq),
+ )
+import Effect.Logger (Logger (..), Pretty (pretty), logDebug)
+import Effect.ReadFS (Has, ReadFS, readContentsYaml)
+import Graphing (Graphing, direct, empty)
+import Path
+import Strategy.Dart.PubSpecLock (PackageName (..))
+import Types (GraphBreadth (..))
+
+data PubSpecContent = PubSpecContent
+  { pubSpecDependencies :: Maybe (Map PackageName PubSpecDepSource)
+  , pubSpecDevDependencies :: Maybe (Map PackageName PubSpecDepSource)
+  , pubSpecDependenciesOverrides :: Maybe (Map PackageName PubSpecDepSource)
+  }
+  deriving (Show, Eq, Ord)
+
+data PubSpecDepSource
+  = PubSpecDepHostedSource
+      { version :: Maybe VerConstraint
+      , hostedName :: Maybe Text
+      , hostedUrl :: Maybe Text
+      }
+  | PubSpecDepGitSource
+      { gitRef :: Maybe VerConstraint
+      , gitUrl :: Text
+      }
+  | PubSpecDepSdkSource {sdkName :: Text}
+  | PubSpecDepPathSource {hostPath :: Text}
+  deriving (Show, Eq, Ord)
+
+instance FromJSON PubSpecContent where
+  parseJSON = Yaml.withObject "pubspec.yaml content" $ \o -> do
+    dependencies <- o .:? "dependencies"
+    devDependencies <- o .:? "dev_dependencies"
+    depOverrides <- o .:? "dependency_overrides"
+    pure $ PubSpecContent dependencies devDependencies depOverrides
+
+-- Formatter crashes here.
+{- ORMOLU_DISABLE -}
```

Comment:
I was able to run the formatter here with no problems, can you repro your crash?  I'd prefer not to disable anything from being formatted.

Reply:
NVM, I tracked it down to your first commit.  You can join everything onto one line to make the formatter try again.  Can we remove this?